ugs-platform/ugs-platform-visualizer/src/main/java/com/willwinder/ugs/nbm/visualizer/renderables/GcodeModel.java

```diff
@@ -314,32 +315,32 @@ private void updateVertexBuffers() {
                     Point3d p1 = ls.getStart();
                     Point3d p2 = ls.getEnd();
 
-                    c[0] = (byte)color.getRed();
-                    c[1] = (byte)color.getGreen();
-                    c[2] = (byte)color.getBlue();
+                    c[0] = (byte) color.getRed();
+                    c[1] = (byte) color.getGreen();
+                    c[2] = (byte) color.getBlue();
 
                     // colors
                     //p1
                     lineColorData[colorIndex++] = c[0];
                     lineColorData[colorIndex++] = c[1];
                     lineColorData[colorIndex++] = c[2];
-                    
+
                     //p2
                     lineColorData[colorIndex++] = c[0];
                     lineColorData[colorIndex++] = c[1];
                     lineColorData[colorIndex++] = c[2];
-                    
+
                     // p1 location
-                    lineVertexData[vertIndex++] = (float)p1.x;
-                    lineVertexData[vertIndex++] = (float)p1.y;
-                    lineVertexData[vertIndex++] = (float)p1.z;
+                    lineVertexData[vertIndex++] = (float) p1.x;
+                    lineVertexData[vertIndex++] = (float) p1.y;
+                    lineVertexData[vertIndex++] = (float) p1.z;
                     //p2
-                    lineVertexData[vertIndex++] = (float)p2.x;
-                    lineVertexData[vertIndex++] = (float)p2.y;
-                    lineVertexData[vertIndex++] = (float)p2.z;
+                    lineVertexData[vertIndex++] = (float) p2.x;
+                    lineVertexData[vertIndex++] = (float) p2.y;
+                    lineVertexData[vertIndex++] = (float) p2.z;
                 }
             }
-            
+
             this.colorArrayDirty = true;
             this.vertexArrayDirty = true;
```

Comment:
Maybe adding these as another early exit condition would fix the race condition? These arrays may only be used when updating the render, which would mean we could be modifying them a second time while the first update is being applied.